### PR TITLE
Improve build speed + Prepare fixing docker tests

### PR DESCRIPTION
**Pull request by michel-tricot**
## What
Improve build speed + prepare fixing docker tests + simplify starting from pre-built image

## How
Stop going through docker to build.
Removed all the non-docker singer artifacts

## Future
Might restore extracting webapp build from gradle.

### CODE REVIEW

**Review by cgardens — APPROVED** [review]
> Might restore extracting webapp build from gradle.

why? otherwise looks good.

**Review by jrhizor — APPROVED** [review]
So the way to think about this is that `gradle` is responsible for running builds and unit tests across all languages, but we lose the ability to know that our docker images build correctly (or at all). 

The integration tests we've talked about so far are running as part of JUnit still, but presumably this means we'll need to add integration tests that use the actual app spun up on docker-compose which will be required for releasing a new version.

So basically this removes any guarantee that `master` is usable on docker-compose for the sake of faster iteration speed? That sounds like a reasonable tradeoff, but we should change our instructions to use `git checkout tags/latest` or similar to guarantee that a new user is starting from a point that is actually integration tested. 

And it speeds up the build to 6min.

All of the Docker parts look much cleaner now 👍
